tests/recordCallsToSetTimeout.ts

```diff
@@ -0,0 +1,193 @@
+    return new Promise<void>((resolve) => {
+      const index = this.calls.findIndex((call) => call.duration === duration);
+
+      if (index === -1) {
+        const listener = (call: SetTimeoutCall, callIndex: number) => {
+          if (call.duration === duration) {
+            call.callback();
+            this.calls.splice(callIndex, 1);
+            this.events.off('setTimeoutAdded', listener);
+            resolve();
+          }
+        };
+        this.events.on('setTimeoutAdded', listener);
+      } else {
+        this.calls[index].callback();
+        this.calls.splice(index, 1);
+        resolve();
+      }
+    });
+  }
+
+  /**
+   * Registers a callback that will be called when `setTimeout` is called and
+   * the expected number of `setTimeout` calls (as specified via
+   * `numAutomaticCalls`) is exceeded.
+   *
+   * @param callback - The callback to register.
+   */
+  onNumAutomaticCallsExhausted(callback: () => void): void {
+    this.events.on('numCallsToPassThroughExhausted', callback);
+  }
+
+  /**
+   * The function with which to replace the global `setTimeout` function. This
+   * mock implementation will record the call to `setTimeout`, along with its
+   * callback and duration, in a call stack, which can be accessed later.
+   *
+   * @param callback - The callback associated with a particular `setTimeout`
+   * call.
+   * @param duration - The duration associated with a particular `setTimeout`
+   * call.
+   * @returns An instance of NodeJS.Timeout which is only supplied to fulfill
+   * the existing type of `setTimeout` and serves no purpose.
+   */
+  _mockSetTimeoutImplementation = (
+    callback: SetTimeoutCallback,
+    duration: number | undefined = 0,
+  ): NodeJS.Timeout => {
+    // Build a "fake" Timeout object
+    const timeout = this.originalSetTimeout(EMPTY_FUNCTION, 0);
+    const interceptedCallback = this.interceptCallback(
+      callback,
+      this._stopPassingThroughCalls,
+    );
+    const call = {
+      callback: interceptedCallback,
+      duration,
+      timeout,
+    };
+    this.calls.push(call);
+
+    if (this.numAutomaticCallsRemaining > 0) {
+      call.callback();
+      this.numAutomaticCallsRemaining -= 1;
+    } else {
+      this.events.emit('numCallsToPassThroughExhausted');
+    }
+    this.events.emit('setTimeoutAdded');
+    return timeout;
+  };
+
+  /**
+   * The function with which to replace the global `clearTimeout` function. This
+   * mock implementation will find a call to `setTimeout` that returned the
+   * given Timeout object and remove it from the queue. If no such call has been
+   * made, then this does nothing.
+   *
+   * @param timeout - A Timeout object as returned by `setTimeout`.
+   */
+  _mockClearTimeoutImplementation = (timeout: NodeJS.Timeout): void => {
+    const index = this.calls.findIndex((c) => c.timeout === timeout);
+
+    if (index !== -1) {
+      this.calls.splice(index, 1);
+    }
+  };
+
+  _stopPassingThroughCalls = () => {
```

Comment:
Nit: this seems like it should be a private field or method

Reply:
That's fair. Changed in [5d8e7b3](https://github.com/MetaMask/eth-block-tracker/pull/103/commits/5d8e7b3cd8f2cc05ee792a7c38496a88c2cf0a04).